src/web/mjs/connectors/Cuutruyen.mjs

```diff
@@ -0,0 +1,117 @@
+        const request = new Request(req, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return new Manga(this, mangaid, data.data.attributes.name.trim());
+    }
+
+    async _getMangas() {
+        const uri = new URL('/api/v2/mangas/recently_updated?page=1&per_page=30', this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pages = data._metadata.total_pages;
+
+        const mangaList = this._getMangasFromPage(data);
+
+        for (let page = 2; page <= pages; page++) {
+            const uri = new URL(`/api/v2/mangas/recently_updated?page=${page}&per_page=30`, this.api);
+            const request = new Request(uri, this.requestOptions);
+            const data = await this.fetchJSON(request);
+            const mangas = await this._getMangasFromPage(data);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    _getMangasFromPage(data) {
+        return data.data.map((element) => {
+            return {
+                id: element.id,
+                title: element.name.trim(),
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const uri = new URL('/api/v2/mangas/' + manga.id + '/chapters', this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        return data.data.map((element) => {
+            return {
+                id: element.id,
+                title: 'Chapter ' + element.number,
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        const uri = new URL('/api/v2/chapters/' + chapter.id, this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pages = data.data.pages.filter((image) => image.status === "processed");
+        return pages.map((image) => {
+            return this.createConnectorURI({
+                url: image.image_url,
+                scrambleString: image.drm_data,
+            });
+        });
+    }
+
+    async _handleConnectorURI(payload) {
+        const image = await new Promise((resolve) => {
+            const image = new Image();
+            image.src = payload.url;
+            image.onload = () => {
+                resolve(image);
+            };
+        });
+        const canvas = await this._descramble(image, payload.scrambleString);
+        const blobFinally = await this._canvasToBlob(canvas);
+        return this._blobToBuffer(blobFinally);
+    }
+
+    async _descramble(image, scrambleString) {
+        await this.wasm;
+
+        let canvas = document.createElement('canvas');
+        canvas.width = image.width;
+        canvas.height = image.height;
+        const ctx = canvas.getContext('2d');
+        ctx.textAlign = "right";
+
+        render_image(image, ctx, scrambleString);
```

Comment:
The website's descrambling doesn't look to difficult. Isn't it possible to reverse engineer their algorithm and just implement it in a single function instead of all these `wasm` overhead?

Reply:
I know a lot of things about reversing PE files but nothing about Wasm :D

The other solution is to use wasm to js "converters" and that a security concern  : you have to trust the JS output

Reply:
> The website's descrambling doesn't look to difficult. Isn't it possible to reverse engineer their algorithm and just implement it in a single function instead of all these `wasm` overhead?

I have tried for months to no avail; there is simply no good tooling for reverse engineering WebAssembly.

Reply:
> I know a lot of things about reversing PE files but nothing about Wasm :D
> 
> The other solution is to use wasm to js "converters" and that a security concern : you have to trust the JS output

There's `wasm2js` from [`binaryen`](https://github.com/WebAssembly/binaryen), a WASM compiler toolchain officially maintained by the WebAssembly working group.